### PR TITLE
Using explosives at end round now gibs you

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Using explosives at end round grief now gibs the user.

If there is no user it just backfires and does nothing. Admins are immune to this if they need to bomb CC for whatever reason.

## Why It's Good For The Game
Gibbing the entire shuttle at EORG is boring.

## Testing
Threw a grenade at round end
image

<hr>

### Declaration
- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/feature_freeze/CODE_OF_CONDUCT.md#currently-changes-to-the-following-types-of-content-reuires-pre-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
<hr>

## Changelog
This is intentionally vague just to see who gets caught out.

:cl: AffectedArc07
tweak: Tweaked explosive behaviour during end of round grief.
/:cl:

### CODE REVIEW

**Review by DGamerL — APPROVED** [review]
Extremely based PR

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -1829,9 +1829,8 @@
 			to_chat(owner, "Standby. Reload cycle in progress. Gunnery crews ready in five seconds!")
 			return
 
-		GLOB.BSACooldown = 1
-		spawn(50)
-			GLOB.BSACooldown = 0
+		GLOB.BSACooldown = TRUE
+		addtimer(VARSET_CALLBACK(GLOB, BSACooldown, FALSE), 5 SECONDS)
```
Comment (lewcc) [inline code/modules/admin/topic.dm:1833]: This kinda feels like an unrelated change for the PR

**Comment by matttheficus** [comment]
Should make the BoH thing make one-tile singulos during End of Round too.

File: code/modules/admin/topic.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/modules/admin/topic.dm:1833]: It kinda is, but while I was at the source of these vars I had to do something about this

**Review by FunnyMan3595 — APPROVED** [review]

File: code/modules/admin/misc_admin_procs.dm [hunk 2]
```diff
@@ -1,5 +1,6 @@
-GLOBAL_VAR_INIT(BSACooldown, 0)
-GLOBAL_VAR_INIT(nologevent, 0)
+GLOBAL_VAR_INIT(BSACooldown, FALSE)
+GLOBAL_VAR_INIT(nologevent, FALSE)
+GLOBAL_VAR_INIT(disable_explosions, FALSE)
```
Comment (Sirryan2002) [inline code/modules/admin/misc_admin_procs.dm:3]: documentttttttt

**Review by Sirryan2002 — APPROVED** [review]

**Comment by Cress67** [comment]
Does this include flashbangs? Because that'd be exceptionally funny.

**Comment by YoureNotTheSharpest** [comment]
lets go

**Comment by AffectedArc07** [comment]
Drafted for these reasons
image

**Comment by Scribble-Sheep** [comment]
Would this trigger on only explosive grenades, or will non-harmful stuff like fliptonium or slipping foam gib you?

**Comment by Coolrune206** [comment]
Does this trigger on arming or detonation?

If it's just on arming, I definitely see people trying to throw their grenades in the seconds before the shuttle docks or using alternative triggers, etc to get around this. Will probably result in them messing up a bit and blowing it up early :y

**Comment by AffectedArc07** [comment]
> Does this trigger on arming or detonation?
> 
> If it's just on arming, I definitely see people trying to throw their grenades in the seconds before the shuttle docks or using alternative triggers, etc to get around this. Will probably result in them messing up a bit and blowing it up early :y

Detonation.

Its not on grenades themselves, its on the explosion function.

**Comment by Coolrune206** [comment]
> > Does this trigger on arming or detonation?
> > If it's just on arming, I definitely see people trying to throw their grenades in the seconds before the shuttle docks or using alternative triggers, etc to get around this. Will probably result in them messing up a bit and blowing it up early :y
> 
> Detonation.
> 
> Its not on grenades themselves, its on the explosion function.

Brutal.

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 3]
```diff
@@ -69,6 +69,14 @@
 		if(response == "Yes")
 			user.visible_message("<span class='warning'>[user] grins as [user.p_they()] begin[user.p_s()] to put a Bag of Holding into a Bag of Holding!</span>", "<span class='warning'>You begin to put the Bag of Holding into the Bag of Holding!</span>")
 			if(do_after(user, 30, target=src))
+				if(GLOB.disable_explosions)
+					if(istype(user, /mob/living/carbon/human))
+						to_chat(user, "<span class='userdanger'>You seem to stuff yourself into the quantum hellscape between the two bags. That wasn't wise.</span>")
+						var/mob/living/carbon/human/H = user
+						H.gib()
```
Comment (DGamerL) [inline code/game/objects/items/weapons/storage/backpack.dm:76]: `gib()` is defined on `/mob`, so we don't need this typecast (but the `istype()` might be good to be replaced by a `QDELETED` check?

Comment (AffectedArc07) [reply code/game/objects/items/weapons/storage/backpack.dm:76]: Are we deeming this a necessary change?

Comment (DGamerL) [reply code/game/objects/items/weapons/storage/backpack.dm:76]: Remove typecast, yes, the rest is fully up to you

**Review by DGamerL — APPROVED** [review]
